modules/kotest/src/main/kotlin/org/testcontainers/containers/Extensions.kt

```diff
@@ -0,0 +1,9 @@
+package org.testcontainers.containers
+
+import io.kotest.core.listeners.TestListener
+import org.testcontainers.lifecycle.Startable
+import org.testcontainers.lifecycle.TestLifecycleAware
+
+fun Startable.perTest(): TestListener = StartablePerTestListener(this)
+fun Startable.perSpec(): TestListener = StartablePerSpecListener(this)
+fun TestLifecycleAware.perTest(): TestListener = TestLifecycleAwareListener(this)
```

Comment:
no need to define an extension method for it. Just before starting or stopping a `Startable` check whether it implements `TestLifecycleAware` and call the appropriate methods

Reply:
Is all TestLifecycleAware are supposed to be startable?
If no does it make sense to call testLifecyleAware method in StartablePerTestListener instead of a separate  TestLifecycleAwareListener?

Reply:
yes, `TestLifecycleAware` implies `Startable`

Reply:
Ok i will do that

Reply:
Done